pallets/validators-manager/src/lib.rs

```diff
@@ -667,6 +678,22 @@ impl<T: Config> NewSessionHandler<T::AuthorityId, T::AccountId> for Pallet<T> {
                         ingress_counter,
                         
                     );
+                } else if validators_action_data.status == ValidatorsActionStatus::AwaitingConfirmation &&
+                validators_action_data.action_type.is_activation() {
```

Comment:
Why do we need this branch here? Was it not working previously?

Reply:
this branch existed before it initiated the activation 
```
else if Self::can_setup_voting_to_activate_validator(
                    &validators_action_data,
                    &action_account_id,
                    active_validators,
                ) {
                    Self::setup_voting_to_activate_validator(
                        ingress_counter,
                        &action_account_id,
                        calculate_two_third_quorum(active_validators.len() as u32),
                        voting_period_end,
                    );
                }
```

so now we just activate the validator